builtin/mainmenu/tab_local.lua

```diff
@@ -379,7 +379,10 @@ local function on_change(type, old_tab, new_tab)
 			gamebar:hide()
 		end
 		core.set_topleft_text("")
-		mm_game_theme.update(new_tab,nil)
+		-- If new_tab is nil, a dialog is being shown; avoid resetting the theme
+		if new_tab then
+			mm_game_theme.update(new_tab,nil)
```

Comment:
hmm not sure about this
can't `update` just avoid playing the song anew?

Reply:
AFAICT the theme does not need to be updated when entering a dialog, and the tab is only hidden when entering a dialog, so the code is logically sound.

All `mm_game_theme.update` would do here is call `mm_game_theme.reset`, which is what stops the music playing. However, it also clears `mm_game_theme.gameid`. `gameid` is checked soon after by `mm_game_theme.update_game`, and if it has been cleared, the music will be restarted anyway. I think it's best to avoid `mm_game_theme.reset` altogether.

Reply:
Ok then